1. 单元测试和测试用例
    name_function.py, names.py
    unittest, 单元测试，测试用例，全覆盖式测试
2. 可通过的测试
    import unittest, 创建一个继承unittest.TestCase 的类, class NamesTestCase(unittest.TestCase), 
    断言，assertEqual, unittest.main(), 
3. 不能通过的测试
    第1行输出只有一个字母E （见❶） ， 它指出测试用例中有一个单元测试导致了错误
4. 测试未通过时怎么办
    测试未通过时，不要修改测试，而应修复导致测试不能通过的代码：检查刚对函数所做的修改，找出导致函数行为不符合预期的修改
5. 添加新测试
    方法名必须以test_打头

# 参考

1. Python编程-从入门到实践.2016 - 18.Django入门
    1. https://github.com/bastamon/Python-Crash-Course